low-level/sensors/impl/platforms/realstor/realstor_logical_volume_sensor.py

```diff
@@ -310,6 +339,90 @@ def _get_msgs_for_faulty_disk_groups(self, disk_groups, send_message=True):
             alert_type = ""
         return faulty_disk_group_messages
 
+    def _get_msgs_for_faulty_logical_volumes(self, logical_volumes, disk_group, send_message=True):
+        """Checks for health of logical volumes and returns list of messages to be
+           sent to handler if there are any.
+        """
+        faulty_logical_volume_messages = []
+        internal_json_msg = None
+        logical_volume_health = None
+        serial_number = None
+        alert_type = ""
+        # Flag to indicate if there is a change in _previously_faulty_logical_volumes
+        state_changed = False
+
+        if not logical_volumes:
+            return
+
+        for logical_volume in logical_volumes:
+            logical_volume_health = logical_volume["health"].lower()
+            serial_number = logical_volume["serial-number"]
+
+            # Check for missing and fault case
+            if logical_volume_health == self.rssencl.HEALTH_FAULT:
+                # Status change from Degraded ==> Fault or OK ==> Fault
```

Comment:
These state transitions may not apply to volumes, as it is from disk group states. confirm possible volume states

Reply:
confirmed that logical volumes too have one of these states - OK, Degraded, Fault, N/A, Unknown.